src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt

```diff
@@ -0,0 +1,152 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.util.Disposer
+import com.intellij.testFramework.EditorTestUtil
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import com.intellij.testFramework.runInEdtAndWait
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.intellij.util.messages.Topic
+import com.sourcegraph.cody.edit.widget.LensAction
+import com.sourcegraph.cody.edit.widget.LensGroupFactory
+import com.sourcegraph.cody.edit.widget.LensLabel
+import com.sourcegraph.cody.edit.widget.LensSpinner
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+
+class DocumentCodeTest : BasePlatformTestCase() {
+
+  override fun setUp() {
+    super.setUp()
+    configureFixture()
+  }
+
+  override fun tearDown() {
+    // TODO: Notify the Agent that all documents were closed.
+    super.tearDown()
+  }
+
+  fun testGetsWorkingGroupLens() {
+    // Do this before starting the edit operation, in order to be subscribed before
+    // the synchronous notification is sent, because it is not buffered or queued anywhere.
+    val foldingRangeFuture = subscribeToTopic(CodyInlineEditActionNotifier.TOPIC_FOLDING_RANGES)
+
+    val editor = myFixture.editor
+    assertFalse(editor.inlayModel.hasBlockElements())
+
+    // Execute the action and await the working group lens.
+    runInEdtAndWait { EditorTestUtil.executeAction(editor, "cody.documentCodeAction") }
+
+    val context = waitForTopic(CodyInlineEditActionNotifier.TOPIC_DISPLAY_WORKING_GROUP)
+    assertNotNull("Timed out waiting for working group lens", context)
+
+    // The inlay should be up.
+    assertTrue("Lens group inlay should be displayed", editor.inlayModel.hasBlockElements())
```

Comment:
...oh I see my question about unsynchronized access was not theoretical. Is `runInEdtAndWait` a fence? But how could it be that if the EditorImpl's inlay model, block elements collection, etc. all unsynchronized, not-volatile stuff?